### PR TITLE
Integrate CUDA sampler into ASR runner and enable skip_copy for decoder

### PR DESCRIPTION
**Pull request by larryliu0820**
- cuda_backend.cpp: Support comma-separated method names in
  skip_copy_output_to_cpu_for_method backend option.
- runner.cpp: Use CudaSampler for argmax when CUDA is available and
  temperature==0. Skip copy to CPU for both encoder and decoder methods.
- CMakeLists.txt updates: Link against extension_llm_sampler_cuda library
  and include the sampler subdirectory.

This optimization keeps decoder logits on GPU and performs argmax directly
on GPU memory, avoiding unnecessary device-to-host copies in the decode loop.

### CODE REVIEW

**Comment by larryliu0820** [comment]
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.13.0) (oldest at bottom):
* __->__ #16388
* #16387
* #16386

File: backends/cuda/runtime/cuda_backend.cpp [hunk 1]
```diff
@@ -83,7 +83,36 @@ class ET_EXPERIMENTAL CudaBackend final
       return false;
     }
     std::lock_guard<std::mutex> guard(skip_copy_method_mutex_);
-    return method_name == skip_copy_method_;
+    // Support comma-separated list of method names
+    if (skip_copy_method_.empty()) {
```
Comment (Gasoonjia) [inline backends/cuda/runtime/cuda_backend.cpp:87]: hmm can we make the `skip_copy_method_` as an array? Maybe we can avoid the fancy comparsion here for supporting multiple method names

**Review by Gasoonjia — APPROVED** [review]

**Comment by larryliu0820** [comment]
obsolete
